app/src/store/actionCreators.js

```diff
@@ -0,0 +1,81 @@
+import { actionTypes } from './actionTypes';
+
+const globalUrl = 'http://localhost:4000';
+
+function getMovies(dispatch, getState){
+    var url = globalUrl + '/movies';
+
+    var state = getState();
+
+    var filter = state.filter.selectedOption.name.toLowerCase();
+    if(filter && filter != 'all'){
+        url += `?filter=${filter}`;
+    }
+
+    var sorter = state.sorter.selectedOption.field.toLowerCase();
+    if(sorter){
+        var sortOrder = state.sorter.asc ? 'asc' : 'desc';
+        url += (filter && filter != 'all' ? '&' : '?') + `sortBy=${sorter}&sortOrder=${sortOrder}`;
+    }
+
+    fetch(url)
+        .then(res => res.json())
+        .then(res => res.data)
+        .then(res => dispatch({
+            type: actionTypes.SET_MOVIES,
+            data: res

```

Comment:
u can skip previous step `.then(res => res.data)`  and use directly res.data

Reply:
Done